### PR TITLE
Update orphaned pod cleaner to use the "FinishedAt" property on the Pod

### PR DESCRIPTION
**Pull request by scme0**
# Background

Rather than creating our own date time, we can use the Finished At property on the Pod we get from the API.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests/Kubernetes/KubernetesOrphanedPodCleanerTests.cs [hunk 1]
```diff
@@ -72,28 +62,15 @@ public async Task OrphanedPodCleanedUpIfOver10MinutesHavePassed()
             await podService.Received().Delete(scriptTicket, Arg.Any<CancellationToken>());
         }
 
-        [TestCase("Succeeded", true)]
-        [TestCase("Failed", true)]
-        [TestCase("Running", false)]
+        [TestCase(PodState.Succeeded, true)]
```
Comment (flin-8) [inline source/Octopus.Tentacle.Tests/Kubernetes/KubernetesOrphanedPodCleanerTests.cs:65]: Add a case for `Pending` state?

Comment (scme0) [reply source/Octopus.Tentacle.Tests/Kubernetes/KubernetesOrphanedPodCleanerTests.cs:65]: The `PodState` enum only has three states:
```csharp
    public enum PodState
    {
        Running,
        Succeeded,
        Failed
    }
```

Comment (scme0) [reply source/Octopus.Tentacle.Tests/Kubernetes/KubernetesOrphanedPodCleanerTests.cs:65]: I see what you mean now - the Phase can have 5 different values.

Comment (scme0) [reply source/Octopus.Tentacle.Tests/Kubernetes/KubernetesOrphanedPodCleanerTests.cs]: I've added additional tests for those phases 👍

**Review by flin-8 — APPROVED** [review]
LGTM

**Pull request merged**